module/chunks/chunkVerifier.go

```diff
@@ -204,17 +205,53 @@ func (fcv *ChunkVerifier) verifyTransactionsInContext(
 	chunkState := fvmState.NewExecutionState(nil, fvmState.DefaultParameters())
 
 	var problematicTx flow.Identifier
-
 	// collect execution data formatted transaction results
+	var txStartIndex int
+	var systemResult *flow.LightTransactionResult
+
+	if systemChunk {
+		transactions, systemResult, err = fcv.createSystemChunk(
+			callbackCtx,
+			&snapshotTree,
+			chunkState,
+			transactionOffset,
+			&events,
+			&serviceEvents,
+			unknownRegTouch,
+			execResID,
+			chIndex,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create system chunk transactions: %w", err)
+		}
+	}
+
 	var txResults []flow.LightTransactionResult
 	if len(transactions) > 0 {
 		txResults = make([]flow.LightTransactionResult, len(transactions))
 	}
 
-	// executes all transactions in this chunk
-	for i, tx := range transactions {
+	// If system chunk, we already executed the process callback transaction so skip it
+	// by setting the start index to 1 and assigning existing process result to tx results
+	if systemResult != nil {
+		txResults[0] = *systemResult
+		txStartIndex = 1
+	}
+
+	// Executes all transactions in this chunk (or remaining transactions for callbacks)
+	for i := txStartIndex; i < len(transactions); i++ {
+		tx := transactions[i]
+		ctx := context
+
+		// For system chunks with callbacks:
+		// - Process callback transaction and callback executions use callbackCtx
+		// - System transaction (last one) uses the original system chunk context
+		if systemChunk && context.ScheduleCallbacksEnabled && i < len(transactions)-1 {
+			ctx = callbackCtx
+		}
```

Comment:
I don't like this the most, but this whole function does way too much and I don't see a nicer way to achieve this beside refactoring the whole function, which I feel it would be too huge a change for this PR especially since it's a critical function. If there's an easy nicer way let me know. Otherwise we might want to refactor this in the future.